File: src/test/java/org/oewntk/xml/in/TestsXmlModelLexGroups.java
/*
 * Copyright (c) 2021-2021. Bernard Bou.
 */

package org.oewntk.xml.in;

import org.junit.BeforeClass;
import org.junit.Test;
import org.oewntk.model.LibTestModelLexGroups;
import org.xml.sax.SAXException;

import javax.xml.parsers.ParserConfigurationException;
import java.io.IOException;

public class TestsXmlModelLexGroups
{
	@BeforeClass
	public static void init() throws IOException, ParserConfigurationException, SAXException
	{
		TestsXmlCommon.init();
	}

	@Test
	public void testCIMultipleAll()
	{
		LibTestModelLexGroups.testCIMultipleAll(TestsXmlCommon.model, TestsXmlCommon.ps);
	}

	@Test
	public void testCILemmas()
	{
		LibTestModelLexGroups.testCILemmas(TestsXmlCommon.model, "battle of verdun", TestsXmlCommon.ps);
	}

	@Test
	public void testCICounts()
	{
		LibTestModelLexGroups.testCICounts(TestsXmlCommon.model, "battle of verdun", TestsXmlCommon.ps);
	}

	@Test
	public void testCICountsFromMap()
	{
		LibTestModelLexGroups.testCICountsFromMap(TestsXmlCommon.model, "battle of verdun", TestsXmlCommon.ps);
	}

	@Test
	public void testCIHypermapWest()
	{
		LibTestModelLexGroups.testCIHypermap(TestsXmlCommon.model, "west", TestsXmlCommon.ps);
	}

	@Test
	public void testCIHypermapBaroque()
	{
		LibTestModelLexGroups.testCIHypermap(TestsXmlCommon.model, "baroque", TestsXmlCommon.ps);
	}

	@Test
	public void testCIAi()
	{
		LibTestModelLexGroups.testCILexesFor(TestsXmlCommon.model, "ai", TestsXmlCommon.ps);
	}

	@Test
	public void testCIBaroque()
	{
		LibTestModelLexGroups.testCILexesFor(TestsXmlCommon.model, "baroque", TestsXmlCommon.ps);
	}

	@Test
	public void testCIWest3()
	{
		LibTestModelLexGroups.testCILexesFor3(TestsXmlCommon.model, "West", TestsXmlCommon.ps);
	}

	@Test
	public void testCIBaroque3()
	{
		LibTestModelLexGroups.testCILexesFor3(TestsXmlCommon.model, "Baroque", TestsXmlCommon.ps);
	}

	@Test
	public void testCIAi3()
	{
		LibTestModelLexGroups.testCILexesFor3(TestsXmlCommon.model, "Ai", TestsXmlCommon.ps);
	}

	@Test
	public void testCIAbsolute3()
	{
		LibTestModelLexGroups.testCILexesFor3(TestsXmlCommon.model, "Absolute", TestsXmlCommon.ps);
	}
}

File: src/test/java/org/oewntk/xml/in/TestsXmlModelLexGroups.kt
/*
 * Copyright (c) 2021-2024. Bernard Bou.
 */
package org.oewntk.xml.`in`

import org.junit.BeforeClass
import org.junit.Test
import org.oewntk.model.Key2
import org.oewntk.model.Lemma
import org.oewntk.model.Lex
import org.oewntk.model.Lex.Groups.groupByLCLemmaThenByKey2
import org.oewntk.model.LibTestModelLexGroups.testCICounts
import org.oewntk.model.LibTestModelLexGroups.testCICountsFromMap
import org.oewntk.model.LibTestModelLexGroups.testCIHypermap
import org.oewntk.model.LibTestModelLexGroups.testCILemmas
import org.oewntk.model.LibTestModelLexGroups.testCILexesFor
import org.oewntk.model.LibTestModelLexGroups.testCILexesFor3
import org.oewntk.model.LibTestModelLexGroups.testCIMultipleAll
import org.oewntk.model.LowerCasedLemma
import org.oewntk.xml.`in`.LibTestsXmlCommon.model

class TestsXmlModelLexGroups {

    @Test
    fun testCIMultipleAll() {
        testCIMultipleAll(model, LibTestsXmlCommon.ps)
    }

    @Test
    fun testCILemmas() {
        testCILemmas(model, Lemma("battle of verdun"), LibTestsXmlCommon.ps)
    }

    @Test
    fun testCICounts() {
        testCICounts(model, Lemma("battle of verdun"), LibTestsXmlCommon.ps)
    }

    @Test
    fun testCICountsFromMap() {
        testCICountsFromMap(model, Lemma("battle of verdun"), LibTestsXmlCommon.ps)
    }

    @Test
    fun testCIAi() {
        testCILexesFor(model, Lemma("ai"), LibTestsXmlCommon.ps)
    }

    @Test
    fun testCIBaroque() {
        testCILexesFor(model, Lemma("baroque"), LibTestsXmlCommon.ps)
    }

    @Test
    fun testCIWest3() {
        testCILexesFor3(model, Lemma("West"), LibTestsXmlCommon.ps)
    }

    @Test
    fun testCIBaroque3() {
        testCILexesFor3(model, Lemma("Baroque"), LibTestsXmlCommon.ps)
    }

    @Test
    fun testCIAi3() {
        testCILexesFor3(model, Lemma("Ai"), LibTestsXmlCommon.ps)
    }

    @Test
    fun testCIAbsolute3() {
        testCILexesFor3(model, Lemma("Absolute"), LibTestsXmlCommon.ps)
    }

    private val lexLCHyperMap: Map<Lemma, Map<Key2, Collection<Lex>>> by lazy { model.lexes.asSequence().groupByLCLemmaThenByKey2() }

    @Test
    fun testCIHypermapWest() {
        testCIHypermap(lexLCHyperMap, Lemma("west"), LibTestsXmlCommon.ps)
    }

    @Test
    fun testCIHypermapBaroque() {
        testCIHypermap(lexLCHyperMap, Lemma("baroque"), LibTestsXmlCommon.ps)
    }

    companion object {

        @JvmStatic
        @BeforeClass
        fun init() {
            model
        }
    }
}
